### ISSUE TITLE
Metadata request for tables fails

### ISSUE THREAD

**Issue by michaelkruglos** [Azure/Azurite#121]
The request is http://127.0.0.1:10002/devstoreaccount1/$metadata#Tables
The resonse is `Cannot read property 'tableName' of undefined`
It should respond correctly to the request above, and shoudn't return such errors.

**Comment by philippgille** [comment]
I have a similar issue, but with a different enpoint.

I start the Docker container like this: `docker run -it --rm -e executable=table -p 10002:10002 arafato/azurite`

Then I'm working with the "Azure SDK for Go" and this is my code:

```go
...
connString := "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;TableEndpoint=http://127.0.0.1:10002/devstoreaccount1;"
storageClient, err := storage.NewClientFromConnectionString(connString)
if err != nil {
	panic(err)
}
tableService := storageClient.GetTableService()
tableServicePtr := &tableService
serviceProps, err := tableServicePtr.GetServiceProperties()
...
```

The last line is a blocking call, during which Docker container prints the following logs:

```
GET /devstoreaccount1/?comp=properties&restype=service 500 0.424 ms - 45
**PANIC** Something unexpected happened! Table Storage Emulator may be in an inconsistent state!
TypeError: Cannot read property 'tableName' of undefined
    at BbPromise.try (/opt/azurite/lib/middleware/table/validation.js:24:48)
    at tryCatcher (/opt/azurite/node_modules/bluebird/js/release/util.js:16:23)
    at Function.Promise.attempt.Promise.try (/opt/azurite/node_modules/bluebird/js/release/method.js:39:29)
    at module.exports (/opt/azurite/lib/middleware/table/validation.js:18:18)
    at Layer.handle [as handle_request] (/opt/azurite/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/opt/azurite/node_modules/express/lib/router/index.js:317:13)
    at /opt/azurite/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/opt/azurite/node_modules/express/lib/router/index.js:335:12)
    at next (/opt/azurite/node_modules/express/lib/router/index.js:275:10)
    at rawParser (/opt/azurite/node_modules/body-parser/lib/types/raw.js:67:7)
    at Layer.handle [as handle_request] (/opt/azurite/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/opt/azurite/node_modules/express/lib/router/index.js:317:13)
    at /opt/azurite/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/opt/azurite/node_modules/express/lib/router/index.js:335:12)
    at next (/opt/azurite/node_modules/express/lib/router/index.js:275:10)
    at logger (/opt/azurite/node_modules/morgan/index.js:144:5)
```

This log entry appears multiple times, maybe because the Azure SDK for Go has some internal retry mechanism or so.

**Comment by philippgille** [comment]
@gpatryk: I see you pushed some commits regarding this issue, which is great! Is it fixed already? I can test my use-case again if you want, I'll just need a Docker image tag of a build where the fix is included.

**Comment by edwin-huber** [comment]
We still need to implement GetServiceProperties support in v3.
Updating labels on this issue.

**Comment by blueww** [comment]
GetServiceProperties on table already released in Azurite 3.14.2.

**Comment by edwin-huber** [comment]
This issue should be resolved in latest version.
Added test to confirm response to JavaScript SDK's GetServiceProperties in #1157
